tests/e2e/utils/data_builder/metrics.py

```diff
@@ -0,0 +1,130 @@
+from typing import Any, Dict, List, Optional
+
+from mypy_boto3_cloudwatch.type_defs import DimensionTypeDef, MetricDataQueryTypeDef
+
+from aws_lambda_powertools.metrics import MetricUnit
+from tests.e2e.utils.data_builder.common import build_random_value
+
+
+def build_metric_query_data(
+    namespace: str,
+    metric_name: str,
+    period: int = 60,
+    stat: str = "Sum",
+    dimensions: Optional[List[DimensionTypeDef]] = None,
```

Comment:
Instead of None can we set default value to empty list and remoce line 16?

Reply:
Not at function signature, because it'd make it a mutable default. https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Reply:
oh, yes makes total sense! Thx